### ISSUE TITLE
Minification mangling html with conditional comments

### ISSUE THREAD

**Issue by collinsauve** [Taritsyn/WebMarkupMin#102]
Looks like in some cases when a the opening and closing tags are in different conditional comments, this library can rewrite the html structure.

This issue was originally discovered in a far more complex html document, but I have been able to create a minimally reproducible test here:

Given this source html:

```
<html>
<body>

<div><table><tr><td>

    <!--[if mso]>
    <div>
        test1
    <![endif]-->
    test2
    <!--[if mso]>
    </div>
    <![endif]-->

    <!--[if mso]>
    <div>
        test1
    <![endif]-->
    test2
    <!--[if mso]>
    </div>
    <![endif]-->
</td></tr></table></div>

</body>
</html>
```

Running 

```
var settings = new HtmlMinificationSettings
{
    RemoveOptionalEndTags = false,
};
var htmlMinifier = new HtmlMinifier(settings);
var minified = htmlMinifier.Minify(html2);
```

Results in:

```
<html><body><div><table><tr><td><!--[if mso]><div>test1 <![endif]--> test2 <!--[if mso]></td></tr></table></div><![endif]--> <!--[if mso]><div>test1 <![endif]--> test2 <!--[if mso]></div><![endif]--></body></html>
```

Notice the `</td></tr></table></div>` closing tags have been moved inside one of the conditionals.  The source html is a valid structure with or without the conditionals, but is not a valid structure after running minifier.

**Comment by Taritsyn** [comment]
Hello, Collin!

A rather complicated structure of HTML document is used here: start and end tags are placed in different conditional comments. Such code can break many HTML minifiers based on parsers (for example, this code is not correctly processed by the Juriy Zaytsev's [HTML Minifier](http://kangax.github.io/html-minifier/), but it causes other type of error).

This error will be very difficult to fix. At the moment, I recommend two ways to solve this problem:

1. Simplify a HTML code.
2. Wrap the fragment of problem code in a [ignoring comment tag](https://github.com/Taritsyn/WebMarkupMin/wiki/Ignoring-fragments-of-markup).

**Comment by collinsauve** [comment]
Hi Andrey.

Thanks for the tip about the ignoring comment tag - I will use that as a work-around until I can find a way to simplify

**Comment by jonyeezs** [comment]
We use a large amount of conditional comments to manage outlook and IE. The workaround of ignoring comment is not scalable.

I believe there's a bug somewhere if the engine is parsing tags outside of comments **into** comments.

**Comment by uhlirradek** [comment]
+1 to this, same problem as @jonyeezs have - anyone going to fix this?

**Comment by Taritsyn** [comment]
Hello!

This error has been fixed in [version 2.13.6](https://github.com/Taritsyn/WebMarkupMin/releases/tag/v2.13.6).
